test/functional/wallet_blank.py

```diff
@@ -0,0 +1,130 @@
+        wallet.importaddress(ADDRESS_BCRT1_UNSPENDABLE)
+        assert_equal(wallet.getwalletinfo()["blank"], False)
+
+    def test_importpubkey(self):
+        if self.options.descriptors:
+            return
+        self.log.info("Test that importpubkey unsets the blank flag")
+        self.nodes[0].createwallet(wallet_name="ipub", disable_private_keys=True)
+        wallet = self.nodes[0].get_wallet_rpc("ipub")
+        assert_equal(wallet.getwalletinfo()["blank"], True)
+
+        eckey = ECKey()
+        eckey.generate()
+
+        wallet.importpubkey(eckey.get_pubkey().get_bytes().hex())
+        assert_equal(wallet.getwalletinfo()["blank"], False)
+
+    def test_importprivkey(self):
+        if self.options.descriptors:
+            return
+        self.log.info("Test that importprivkey unsets the blank flag")
+        self.nodes[0].createwallet(wallet_name="ipriv", blank=True)
+        wallet = self.nodes[0].get_wallet_rpc("ipriv")
+        assert_equal(wallet.getwalletinfo()["blank"], True)
+        wallet.importprivkey(generate_wif_key())
+        assert_equal(wallet.getwalletinfo()["blank"], False)
+
+    def test_importmulti(self):
+        if self.options.descriptors:
+            return
+        self.log.info("Test that importmulti unsets the blank flag")
+        self.nodes[0].createwallet(wallet_name="imulti", disable_private_keys=True)
+        wallet = self.nodes[0].get_wallet_rpc("imulti")
+        assert_equal(wallet.getwalletinfo()["blank"], True)
+        wallet.importmulti([{
+            "desc": ADDRESS_BCRT1_UNSPENDABLE_DESCRIPTOR,
+            "timestamp": "now",
+        }])
+        assert_equal(wallet.getwalletinfo()["blank"], False)
+
+    def test_importdescriptors(self):
+        if not self.options.descriptors:
+            return
+        self.log.info("Test that importdescriptors unsets the blank flag")
+        self.nodes[0].createwallet(wallet_name="idesc", disable_private_keys=True)
+        wallet = self.nodes[0].get_wallet_rpc("idesc")
+        assert_equal(wallet.getwalletinfo()["blank"], True)
+        wallet.importdescriptors([{
+            "desc": ADDRESS_BCRT1_UNSPENDABLE_DESCRIPTOR,
+            "timestamp": "now",
+        }])
+        assert_equal(wallet.getwalletinfo()["blank"], False)
+
+    def test_importwallet(self):
+        if self.options.descriptors:
+            return
+        self.log.info("Test that importwallet unsets the blank flag")
+        def_wallet = self.nodes[0].get_wallet_rpc(self.default_wallet_name)
+
+        self.nodes[0].createwallet(wallet_name="iwallet", blank=True)
+        wallet = self.nodes[0].get_wallet_rpc("iwallet")
+        assert_equal(wallet.getwalletinfo()["blank"], True)
+
+        wallet_dump_path = os.path.join(self.nodes[0].datadir, "wallet.dump")
+        def_wallet.dumpwallet(wallet_dump_path)
+
+        wallet.importwallet(wallet_dump_path)
+        assert_equal(wallet.getwalletinfo()["blank"], False)
```

Comment:
You are explicitly expecting this behavior, but I’m kinda surprised that when we import a wallet that was explicitly created “blank” after the import it is now no longer blank. As far as I can tell, no keys were created, unless the import automatically does that. I’m curious why this is the expected behavior (although since you seem to explicitly expect that behavior, I don’t think there is an issue here).

Reply:
We determine whether a wallet is new by checking whether it has any keys or scripts contained in it. If it is "new", then the wallet will be setup to have the automatically generated things. The blank flag is used only to indicate to that code that the wallet is intentionally blank and should not be setup as a new wallet. Once something has been added to the wallet, it's no longer blank and the setup code will no longer recognize it as being "new", so the flag should be removed.

Reply:
I see, thanks for the explanation

Reply:
I have a different perspective. The flag was provided during wallet creation and captures the intent of the user.

> The blank flag is used only to indicate to that code that the wallet is intentionally blank and should not be setup as a new wallet.

I'd like to take this one step further and argue that "The blank flag is used only to indicate to that code that the wallet is intentionally blank and **the user is responsible for managing the keys in this wallet manually**." If user created wallet intentionally blank so the software doesn't touch the keys, does the intention go away after the keys are imported in the wallet? I don't think so.

> Once something has been added to the wallet, it's no longer blank and the setup code will no longer recognize it as being "new", so the flag should be removed.

We shouldn't casually flip between just `blank` and `intentionally blank`. Yes, it's not longer blank, but it's still **intentionally blank** because this was and is the intent of the user. I think the word blank is just misleading here. We're better off replacing `intentionally blank` with `manual`.